### PR TITLE
Hybrid Resize

### PR DESCRIPTION
**Pull request by varunagrawal**
This PR adds tests and an overloaded method for resize in `HybridFactorGraph`s.

cc @dellaert

### CODE REVIEW

**Review by ProfFan — COMMENTED** [review]
I think this is kind of not correct? May want to discuss about the real meanings of these sizes first, since I think the size of the FG should be the sum of all sub-fg s?

**Comment by varunagrawal** [comment]
That's what even I though but since we do a `Base::push_back` each time, the factors are retained by the base class even if we delete them from the subclass. Which is why I first added the unit tests to check my assumptions and then implemented the resize method accordingly.

File: gtsam/hybrid/GaussianHybridFactorGraph.cpp [hunk 1]
```diff
@@ -37,7 +37,8 @@ template class EliminateableFactorGraph<GaussianHybridFactorGraph>;
 
 void GaussianHybridFactorGraph::print(
     const string& str, const gtsam::KeyFormatter& keyFormatter) const {
-  Base::print(str, keyFormatter);
```
Comment (varunagrawal) [inline gtsam/hybrid/GaussianHybridFactorGraph.cpp:40]: Everything in this file is formatting.

File: gtsam/hybrid/IncrementalHybrid.cpp [hunk 2]
```diff
@@ -18,8 +18,9 @@
  */
 
 #include <gtsam/hybrid/IncrementalHybrid.h>
-#include <unordered_set>
```
Comment (varunagrawal) [inline gtsam/hybrid/IncrementalHybrid.cpp:21]: Again, mainly formatting here.

**Comment by ProfFan** [comment]
And I think this is due to FactorGraph being public and also the reason why Kevin made it protected.

**Comment by dellaert** [comment]
> And I think this is due to FactorGraph being public and also the reason why Kevin made it protected.

I know - but we want to make HFG a *real* FG. If methods from base class are not appropriate, theys hould be deleted or made to work.

**Review by dellaert — APPROVED** [review]
Approve to move things forward although I think this PR is mainly orthogonal to the paper. We should never use resize inn the first place

**Comment by varunagrawal** [comment]
> Approve to move things forward although I think this PR is mainly orthogonal to the paper. We should never use resize inn the first place

I agree. I made this PR before our discussion yesterday about just creating new objects.

**Pull request merged**